tools/toolchain.cmake.export

```diff
@@ -1,5 +1,6 @@
 set(CMAKE_SYSTEM_NAME Generic)
 set(CMAKE_SYSTEM_VERSION 1)
+set(NUTTX 1)
```

Comment:
should we __NuttX__

Reply:
Usually CMake variables are uppercase but it is possible to mix uppercase and lower. What do you prefer ? (follow CMake style or follow NuttX style ?)

Reply:
I made this suggestion because `__NuttX__` is used in c/c++ code to check nuttx environment:
https://github.com/apache/nuttx/blob/master/tools/Config.mk#L640
https://github.com/apache/nuttx/blob/master/CMakeLists.txt#L500
BTW, Makefile need update too.

Reply:
I think full uppercase is better because it follows CMake style: if you look in ```/usr/share/cmake-xx/Modules/Platform```, you can see that all OS variables are set with uppercase only (i.e. BeOS -> BEOS, Android -> ANDROID, SerenityOS -> SERENITYOS, and so on).